### PR TITLE
[testing] Switch to CodeQL from LGTM

### PR DESCRIPTION
**Pull request by TurboTurtle**
LGTM is shutting down shortly, as it has been folded into GH CodeQL. As such, we need to switch over to using CodeQL to maintain automatic code analysis.

Do this by using the provided CodeQL workflow configuration, and add a config file to limit the analysis to the appropriate `sos/` directory.

Signed-off-by: Jake Hunsaker <jhunsake@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by TurboTurtle** [comment]
The config file seems to be correct now to restrict scanning to `sos/` and not include our test suite. However, there's still a lot of new warnings that LGTM was not presenting to us.

I think with this PR, I'll make another update to tackle the 2 alerts marked as 'high' - from there we can whittle down the new alerts over time.

File: sos/report/plugins/openstack_instack.py [hunk 1]
```diff
@@ -81,16 +81,17 @@ def setup(self):
...
                 self.add_cmd_output([
-                    "openstack stack show %s" % sid[0],
-                    "openstack stack resource list -n 10 %s" % sid[0]
+                    "openstack stack show %s" % sid,
+                    "openstack stack resource list -n 10 %s" % sid
                 ])
 
                 # get details on failed deployments
-                cmd = "openstack stack resource list -f value -n 5 %s" % sid[0]
+                cmd = "openstack stack resource list -f value -n 5 %s" % sid
```
Comment (pmoravec) [inline sos/report/plugins/openstack_instack.py:94]: Why the change from `sid[0]` to `sid[1]` value in commands arguments?

Comment (TurboTurtle) [reply sos/report/plugins/openstack_instack.py:94]: The new regex groups the matches differently - I grabbed some output from some docs and built against, but it would be great if we could have some "real" validation on it.

Comment (pmoravec) [reply sos/report/plugins/openstack_instack.py:94]: Ah I see - but why `re.search` (that searches for the first instance only) when we need to spot all stack IDs?

Moreover, the code doesnt work for me on an an example output like:

```
+--------------------------------------+------------+-----------------+---------------------+--------------+
| ID                                   | Stack Name | Stack Status    | Creation Time       | Updated Time |
+--------------------------------------+------------+-----------------+---------------------+--------------+
| 70b9feca-8f99-418e-b2f1-cc38d61b3ffb | MYSTACK    | CREATE_COMPLETE | 2016-06-08T09:54:15 | None         |
| 82f75a62-9e03-4031-afb1-a63022f8dd68 | SOMESTACK  | ALSO_COMPLETE   | 2022-11-21T08:46:01 | None         |
+--------------------------------------+------------+-----------------+---------------------+--------------+
```

where the `for sid in stack_ids:` fails with `TypeError: '_sre.SRE_Match' object is not iterable`.

Comment (TurboTurtle) [reply sos/report/plugins/openstack_instack.py:94]: Good catch. I was overlooking this was scanning the entire output, not line by line like our other matching methods, so this should go back to `findall()`.

**Review by pmoravec — APPROVED** [review]
ACK to the current changes.

Many CodeQL warnings are (still) raised, we can fix them later on (or within this PR).
